join/slab_join.cpp

```diff
@@ -0,0 +1,165 @@
+#include "slab_join.hpp"
+#include "common/dpcpp/slab_hash.hpp"
+#include <math.h>
+#include "join_helpers/join_helpers.hpp"
+
+using std::pair;
+using namespace join_helpers;
+SlabJoin::SlabJoin() : Dwarf("SlabJoin") {}
+
+void SlabJoin::_run(const size_t buf_size, Meter &meter) {
+  const int scale = 16;
+  auto opts = meter.opts();
+
+  // todo: sizes
+  const std::vector<uint32_t> table_a_keys =
+      helpers::make_unique_random(buf_size);
+  const std::vector<uint32_t> table_a_values =
+      helpers::make_unique_random(table_a_keys.size());
+
+  const std::vector<uint32_t> table_b_keys =
+      helpers::make_unique_random(buf_size);
+  const std::vector<uint32_t> table_b_values =
+      helpers::make_unique_random(table_b_keys.size());
+
+  auto sel = get_device_selector(opts);
+  sycl::queue q{*sel};
+  std::cout << "Selected device: "
+            << q.get_device().get_info<sycl::info::device::name>() << "\n";
+
+  // hash table
+
+  auto expected = join_helpers::seq_join(table_a_keys, table_a_values,
+                                         table_b_keys, table_b_values);
+
+  std::cout << "Expected done\n";
+
+  for (auto it = 0; it < opts.iterations; ++it) {
+    int work_size = ceil((float)buf_size / scale);
```

Comment:
`work_size` isn't really descriptive along with the `scale`, what do you mean by that?

Reply:
Do you mean that naming does not carry information about this variable?